lib/utils.pm

```diff
@@ -226,21 +226,20 @@ Die, if this is not the case.
 
 =cut
 sub integration_services_check_ip {
-    # Workaround for poo#44771 "Can't call method "exec" on an undefined value"
-    select_console('svirt');
-    select_console('sut');
+    my ($keep_open) = shift // 1;
```

Comment:
Not sure why you need this option. If you are sure, that all commands which are called here don't need the same session/environment with others calls, then I don't see any useful thing to make it variable.

Reply:
> Agreed, and on top of this maybe we should remove this [workaround](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/utils.pm#L229) and check whether the test still works on hyperv and vmware

This has already removed

Reply:
ok, haven't check the left side. ty

Reply:
So let's remove the argument and we can merge it.

Reply:
> Not sure why you need this option. If you are sure, that all commands which are called here don't need the same session/environment with others calls, then I don't see any useful thing to make it variable.

Sorry man. i try to understand that. we pass the `keep_open` in `integration_services_check`. which comes to `integration_services_check` and calls the `integration_services_check_ip`. Right now i havent checked everything but we change the behavior only for `integration_services.pm`. Which means that if it called from somewhere else we do not force restart. If i understand you want to make `integration_services_check_ip` have always `keep_open => 0`  for any case. is that right? 
If so i dont think it needs to change. Do not need to force a new connection if the scenario works, right?

Reply:
> > Not sure why you need this option. If you are sure, that all commands which are called here don't need the same session/environment with others calls, then I don't see any useful thing to make it variable.
> 
> Sorry man. i try to understand that. we pass the `keep_open` in `integration_services_check`. which comes to `integration_services_check` and calls the `integration_services_check_ip`. Right now i havent checked everything but we change the behavior only for `integration_services.pm`. Which means that if it called from somewhere else we do not force 
restart. If i understand you want to make `integration_services_check_ip` have always `keep_open => 0` for any case. is that right?
> If so i dont think it needs to change. Do not need to force a new connection if the scenario works, right?

It is used only here in `integration_services_check` and runs for vmware and hyperv as they come with feature called integrated services. Apparently the update in `new_ssh_connection` made a difference and the tests are passing atm. So we can replace `get_cmd_output` but as of now that is it. 
If there is a difference between hyperv and vmware iy should be addressed in the specific branch in `integration_services_check_ip`.

Reply:
[ sle-15-SP3-JeOS-for-MS-HyperV-x86_64-Build2.31-jeos-filesystem@svirt-hyperv-uefi](https://openqa.suse.de/tests/5961247/file/autoinst-log.txt):

```
[2021-05-06T17:49:53.986 CEST] [debug] <<< backend::baseclass::run_ssh_cmd(cmd="powershell -Command \"Get-VM openQA-SUT-20 | Get-VMNetworkAdapter | Format-Table -HideTableHeaders IPAddresses\"", hostname="win2k19.qa.suse.cz", wantarray=1, username=undef, keep_open=1, password="SECRET")
[2021-05-06T17:49:53.986 CEST] [debug] <<< backend::baseclass::run_ssh(cmd="powershell -Command \"Get-VM openQA-SUT-20 | Get-VMNetworkAdapter | Format-Table -HideTableHeaders IPAddresses\"", wantarray=1, hostname="win2k19.qa.suse.cz", keep_open=1, username=undef, password="SECRET")
[2021-05-06T17:49:53.986 CEST] [debug] <<< backend::baseclass::new_ssh_connection(username=undef, password="SECRET", keep_open=1, hostname="win2k19.qa.suse.cz", wantarray=1, blocking=1)
[37m[2021-05-06T17:49:54.018 CEST] [debug] Use existing SSH connection (key:hostname=win2k19.qa.suse.cz,username=root)
[0m[37m[2021-05-06T17:49:56.226 CEST] [debug] [run_ssh_cmd(powershell -Command "Get-VM openQA-SUT-20 | Get-VMNetworkAdapter | Format-Table -HideTableHeaders IPAddresses")] stdout:

{10.100.103.140, fe80::215:5dff:fe11:3e5}
```

and  [sle-15-SP3-JeOS-for-VMware-x86_64-Build2.35-jeos-filesystem@svirt-vmware65](https://openqa.suse.de/tests/5981031/file/autoinst-log.txt)

```
[2021-05-10T09:28:23.711 CEST] [debug] <<< backend::baseclass::run_ssh_cmd(cmd="set -x; vmid=\$(vim-cmd vmsvc/getallvms | awk '/openQA-SUT-1/ { print \$1 }');if [ \$vmid ]; then vim-cmd vmsvc/get.guest \$vmid | awk '/ipAddress/ {print \$3}' | head -n1 | sed -e 's/\"//g' | sed -e 's/,//g'; fi", wantarray=1, username="root", keep_open=1, hostname="openqaw8-vmware.qa.suse.de", password="SECRET")
[2021-05-10T09:28:23.711 CEST] [debug] <<< backend::baseclass::run_ssh(cmd="set -x; vmid=\$(vim-cmd vmsvc/getallvms | awk '/openQA-SUT-1/ { print \$1 }');if [ \$vmid ]; then vim-cmd vmsvc/get.guest \$vmid | awk '/ipAddress/ {print \$3}' | head -n1 | sed -e 's/\"//g' | sed -e 's/,//g'; fi", wantarray=1, username="root", keep_open=1, hostname="openqaw8-vmware.qa.suse.de", password="SECRET")
[2021-05-10T09:28:23.711 CEST] [debug] <<< backend::baseclass::new_ssh_connection(hostname="openqaw8-vmware.qa.suse.de", password="SECRET", username="root", wantarray=1, keep_open=1, blocking=1)
[37m[2021-05-10T09:28:23.711 CEST] [debug] Close broken SSH connection (key:hostname=openqaw8-vmware.qa.suse.de,username=root)
[0m[37m[2021-05-10T09:28:23.930 CEST] [debug] SSH connection to root@openqaw8-vmware.qa.suse.de established
[0m[37m[2021-05-10T09:28:24.474 CEST] [debug] [run_ssh_cmd(set -x; vmid=$(vim-cmd vmsvc/getallvms | awk '/openQA-SUT-1/ { print $1 }');if [ $vmid ]; then vim-cmd vmsvc/get.guest $vmid | awk '/ipAddress/ {print $3}' | head -n1 | sed -e 's/"//g' | sed -e 's/,//g'; fi)] stdout:
10.162.30.251
```

Reply:
> > > > Not sure why you need this option. If you are sure, that all commands which are called here don't need the same session/environment with others calls, then I don't see any useful thing to make it variable.
> > > 
> > > 
> > > Sorry man. i try to understand that. we pass the `keep_open` in `integration_services_check`. which comes to `integration_services_check` and calls the `integration_services_check_ip`. Right now i havent checked everything but we change the behavior only for `integration_services.pm`. Which means that if it called from somewhere else we do not force
> > > restart. If i understand you want to make `integration_services_check_ip` have always `keep_open => 0` for any case. is that right?
> > > If so i dont think it needs to change. Do not need to force a new connection if the scenario works, right?
> > 
> > 
> > It is used only here in `integration_services_check` and runs for vmware and hyperv as they come with feature called integrated services. Apparently the update in `new_ssh_connection` made a difference and the tests are passing atm. So we can replace `get_cmd_output` but as of now that is it and remove the workaround ofc.
> 
> first where is the workaround? or is there another somewhere else?
> second i am confused here? So we do not want to have this implementation and do not set `keep_open` or what?
> Atm the `integration_services_check_ip` is used only by `integration_services_check` but i am thinking that it could be used by others as well. and they would like to have the option to pass `keep_open` for one or the other reason
> 
> > If there is a difference between hyperv and vmware it should be addressed in the specific branch in `integration_services_check_ip`.

Hm... I am just saying that it is enough to drop obsolete get_cmd_output and workaround in this PR. If there will be need to introduce a parameter it can be easily updated, as of now it is good enough to keep the same behaviour as we can see in current test runs. To me, it seems more robust solution to use `keep_open` in both cases as we have now.

Reply:
resolved again as i removed the keep_open as it was demanded